modules/core/src/main/scala/com/azavea/stac4s/ProductFieldNames.scala

```diff
@@ -0,0 +1,22 @@
+package com.azavea.stac4s
+
+import shapeless.ops.hlist.ToTraversable
+import shapeless.ops.record.Keys
+import shapeless.{HList, LabelledGeneric}
+
+trait ProductFieldNames[T] {
+  def get: Set[String]
+}
+
+object ProductFieldNames {
+  def apply[T](implicit ev: ProductFieldNames[T]): ProductFieldNames[T] = ev
+
+  @SuppressWarnings(Array("all"))
```

Comment:
Supresses scapegoats unused argument failure (LabelledGeneric.Aux[*, L] is not used)